systems/subscriber/registry/pkg/server/subscriber.go

```diff
@@ -37,59 +36,50 @@ type SubcriberServer struct {
 	subscriberRoutingKey msgbus.RoutingKeyBuilder
 	pb.UnimplementedRegistryServiceServer
 	simManagerService client.SimManagerClientProvider
-	networkClient     creg.NetworkClient
+	orgId             string
+	orgClient         cnucl.OrgClient
 }
 
-func NewSubscriberServer(orgName string, subscriberRepo db.SubscriberRepo, msgBus mb.MsgBusServiceClient, simManagerService client.SimManagerClientProvider, networkClient creg.NetworkClient) *SubcriberServer {
+func NewSubscriberServer(orgName string, subscriberRepo db.SubscriberRepo, msgBus mb.MsgBusServiceClient, simManagerService client.SimManagerClientProvider, orgId string, orgService cnucl.OrgClient) *SubcriberServer {
 	return &SubcriberServer{
 		orgName:              orgName,
 		subscriberRepo:       subscriberRepo,
 		msgbus:               msgBus,
 		simManagerService:    simManagerService,
-		networkClient:        networkClient,
 		subscriberRoutingKey: msgbus.NewRoutingKeyBuilder().SetCloudSource().SetSystem(pkg.SystemName).SetOrgName(orgName).SetService(pkg.ServiceName),
+		orgId:                orgId,
+		orgClient:            orgService,
 	}
 }
 
 func (s *SubcriberServer) Add(ctx context.Context, req *pb.AddSubscriberRequest) (*pb.AddSubscriberResponse, error) {
 	log.Infof("Adding subscriber: %v", req)
 
 	var dob string
-
+	subscriberId := uuid.NewV4()
 	networkId, err := uuid.FromString(req.GetNetworkId())
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument,
 			"invalid format of network uuid. Error %s", err.Error())
 	}
 
-	orgId, err := uuid.FromString(req.GetOrgId())
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument,
-			"invalid format of org uuid. Error %s", err.Error())
-	}
-
 	if req.GetDob() != "" {
 		dob, err = validate.ValidateDate(req.GetDob())
 		if err != nil {
 			return nil, status.Errorf(codes.InvalidArgument, err.Error())
 		}
 	}
-
-	subscriberId := uuid.NewV4()
-
-	networkInfo, err := s.networkClient.Get(networkId.String())
```

Comment:
No. we still need to verify the provided networkId against available networks for the org, so this line is correct1